### PR TITLE
PP-92: SDK Dapp review

### PR DESCRIPTION
**Pull request by antomor**
## What

- Added dockerfile to deploy the dapp on testnet
- Update README.md
- Remove useless comments
- Replace yarn.lock with package-lock.json

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## relay-dapp dockerfile (This file is called Dockerfile) on server side. ##
+FROM node:12-alpine AS compiler
+RUN apk add --no-cache build-base git bash
+WORKDIR /usr/src/app
+COPY . ./
+RUN npm i
+RUN npm start
```
Comment (mortelli) [inline Dockerfile:7]: should we add documentation somewhere on how to use this? maybe something from the [post-deployment notes](https://hackmd.io/@antomor/Syfk6wAdt)?

File: README.md [hunk 2]
```diff
@@ -1,52 +1,60 @@
 # RIF Relaying Services SDK sample dApp
 
-This a basic sample dApp to showcase how users can submit transactions to the RSK blockchain paying a RelayServer for the transaction fees with tokens. You will need to connect to the dApp with Metamask but only for signing transactions with the account that controls your Smart Wallets.
+This is a sample dApp to showcase how users can submit relayed transactions to the RSK blockchain using RIF Relay SDK. You will need to connect to the dApp with Metamask but only for signing transactions with the account that controls your Smart Wallets.
```
Comment (mortelli) [inline README.md]:
```suggestion
This is a sample dApp to showcase how users can submit relayed transactions to the RSK blockchain using the [RIF Relay SDK](https://github.com/infuy/relaying-services-sdk). You will need to connect to the dApp with MetaMask but only for signing transactions with the account that controls your Smart Wallets.
```

Comment (mortelli) [inline README.md]: as a reader, it's not clear to me what "account that controls your Smart Wallets" means

File: README.md [hunk 3]
```diff
@@ -1,52 +1,60 @@
...
-## Using the dApp
 
 ## Pre-Requisites
 
-* RSKj Node Running
-* [Enveloping Contracts](https://github.com/anarancio/rif-relay-contracts) deployed.
-* [Enveloping Server](https://github.com/infuy/rif-relay-server) running.
+* [NodeJS Version v12.18.3 or higher](https://nodejs.org/en/download/).
+* [RSKj Node Running](https://github.com/rsksmart/rskj).
+* [RIF Relay Contract Contracts](https://github.com/anarancio/rif-relay-contracts) deployed.
```
Comment (mortelli) [inline README.md]:
```suggestion
* [RIF Relay Contracts](https://github.com/anarancio/rif-relay-contracts) deployed.
```

File: README.md [hunk 4]
```diff
@@ -1,52 +1,60 @@
...
+* [Current Token Allowed](https://github.com/anarancio/rif-relay-contracts#allowing-tokens).
```
Comment (mortelli) [inline README.md]:
```suggestion
* [An allowed Token](https://github.com/anarancio/rif-relay-contracts#allowing-tokens).
```

File: README.md [hunk 5]
```diff
@@ -1,52 +1,60 @@
...
-
+1. Edit the sample dApp config file located in `.env` and add the contract addresses. By default the project imports `.env` on the `npm start`. Look for these lines and modify them accordingly:
+```
+REACT_APP_CONTRACTS_RELAY_HUB=0xE0825f57Dd05Ef62FF731c27222A86E104CC4Cad
+REACT_APP_CONTRACTS_DEPLOY_VERIFIER=0x1eD614cd3443EFd9c70F04b6d777aed947A4b0c4
+REACT_APP_CONTRACTS_RELAY_VERIFIER=0x5159345aaB821172e795d56274D0f5FDFdC6aBD9
+REACT_APP_CONTRACTS_SMART_WALLET_FACTORY=0x03F23ae1917722d5A27a2Ea0Bcc98725a2a2a49a
+REACT_APP_CONTRACTS_SMART_WALLET=0x73ec81da0C72DD112e06c09A6ec03B5544d26F05
+REACT_APP_CONTRACTS_RIF_TOKEN=0x726ECC75d5D51356AA4d0a5B648790cC345985ED
+REACT_APP_CONTRACTS_RELAY_WORKER=0x3d67f029f778a088904f12d030933967d220faa3
```
Comment (mortelli) [inline README.md:33]: we should add a note specifying that this address can be obtained from the `getAddr` server endpoint

File: src/contracts/IForwarder.json [hunk 6]
```diff
@@ -0,0 +1,218 @@
+{
```
Comment (mortelli) [inline src/contracts/IForwarder.json:1]: i must have missed this addition until now, i'm a bit surprised we could do without it before [the original PR](https://github.com/wilsoniovlabs/relaying-services-sdk-dapp/pull/1).

any insights on this?

**Review by mortelli — COMMENTED** [review]
thanks for tying up these loose ends. i've left some minor comments.

although all of our manual tests in the last month or 2 have been based on these changes, i still want to run this specific branch and make sure everything works/looks OK one last time before merging.

i'll do this as soon as my open threads are addressed and we can swiftly close this if no issues arise

File: README.md (hunk 5 above)
Comment (antomor) [reply README.md:33]: Yes, definitely. Thank you for pointing out

File: src/contracts/IForwarder.json (hunk 6 above)
Comment (antomor) [reply src/contracts/IForwarder.json:1]: I think it is used in the execute modal and it was already there in the original PR as well.

File: Dockerfile (hunk 1 above)
Comment (antomor) [reply Dockerfile:7]: After discussing this with @rafaiovlabs , I can confirm that this `Dockerfile` isn't meant to be used standalone, but with the `docker-compose` file that will be included in the [rif-relay-server repo](https://github.com/infuy/rif-relay-server) soon.

File: src/contracts/IForwarder.json (hunk 6 above)
Comment (mortelli) [reply src/contracts/IForwarder.json:1]: > I think it is used in the execute modal and it was already there in the original PR as well.

yes, i did see it there, i'm just surprised we didn't need it before that :slightly_smiling_face: 

anyway, resolving, thanks

File: README.md (hunk 2 above)
Comment (mortelli) [reply README.md]: maybe "with the account that will interact with the smart wallet"?

the metamask account doesn't really control everything, no?

lmk what you think @antomor

File: Dockerfile (hunk 1 above)
Comment (mortelli) [reply Dockerfile:7]: should we wait for some update from rafa before merging?

Comment (antomor) [reply Dockerfile:7]: No, I'd merge it as it is and eventually we can integrate later on

File: README.md (hunk 2 above)
Comment (antomor) [reply README.md]: Yes, I think that the most appropriate term to define that account is `owner`, so I replace that sentence with `the account that owns the smart wallet`

**Review by mortelli — APPROVED** [review]
thanks for addressing everything :star:

**Pull request merged**